From akpm@osdl.org Tue Aug  9 00:34:27 2005
Message-Id: <200508090732.j797W1xB025438@shell0.pdx.osdl.net>
From: Andrew Morton <akpm@osdl.org>
Subject: PCI: fix quirk-6700-fix.patch
Date: Tue, 09 Aug 2005 00:30:51 -0700


drivers/built-in.o(.text+0x32c3): In function `quirk_pcie_pxh':
/usr/src/25/drivers/pci/quirks.c:1312: undefined reference to `disable_msi_mode'

Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/pci.h |    6 ++++++
 1 files changed, 6 insertions(+)

--- gregkh-2.6.orig/drivers/pci/pci.h	2005-08-09 15:46:55.000000000 -0700
+++ gregkh-2.6/drivers/pci/pci.h	2005-08-09 16:34:58.000000000 -0700
@@ -46,7 +46,13 @@
 #else
 #define pci_msi_quirk 0
 #endif
+
+#ifdef CONFIG_PCI_MSI
 void disable_msi_mode(struct pci_dev *dev, int pos, int type);
+#else
+static inline void disable_msi_mode(struct pci_dev *dev, int pos, int type) { }
+#endif
+
 extern int pcie_mch_quirk;
 extern struct device_attribute pci_dev_attrs[];
 extern struct class_device_attribute class_device_attr_cpuaffinity;
